src/cli/commands/tasks/tasks-command.ts

```diff
@@ -39,101 +39,115 @@ export class TasksCommand {
     const orderedStructIds: number[] = [];
-
+    const categoryEnum = await this.enumService.getEnum(3413); // Preload category enum
+  
     const difficultyEnums: number[] = [3981, 3982, 3983, 3984, 3985, 3986];
     for (let enumId of difficultyEnums) {
       const orderedDifficultyStructIds: Map<number, string | number> = (await this.enumService.getEnum(enumId)).map;
       for (const structId of orderedDifficultyStructIds.values()) {
         orderedStructIds.push(structId as number);
       }
     }
-
+  
     const allTaskStructs: Struct[] = [];
     for (let structId of orderedStructIds) {
       const taskStruct: Struct = await this.structService.getStruct(structId);
       allTaskStructs.push(taskStruct);
     }
-
-    let allTasksFormatted: any[] = [];
-
-    if (options.legacy) {
-      allTasksFormatted = allTaskStructs.map((s, i) => {
-        const out = {
-          id: '' + (s.params.get(PARAM_ID.CA_VARBIT_INDEX) as number),
-          monster: '' + (s.params.get(PARAM_ID.CA_MONSTER_ID) as number),
-          name: s.params.get(PARAM_ID.CA_NAME) as string,
-          description: s.params.get(PARAM_ID.CA_DESCRIPTION) as string,
-          category: '',
-          tier: this.getLegacyTier(s.params.get(PARAM_ID.CA_TIER_ID) as number),
-          clientSortId: '' + i,
-        };
-        return out;
-      });
-    } else {
-      allTasksFormatted = allTaskStructs.map((s, i) => {
-        const out: ITask = {
-          structId: s.id,
-          sortId: i,
-        };
-        return out;
-      });
-    }
-
+  
+    const allTasksFormatted: ITask[] = allTaskStructs.map((s, i) => {
+      const categoryValue = s.params.get(PARAM_ID.CA_CATEGORY_ID) as number;
+  
+      return {
+        structId: s.id, 
+        sortId: i,      
+        id: '' + (s.params.get(PARAM_ID.CA_VARBIT_INDEX) as number),
+        monster: '' + (s.params.get(PARAM_ID.CA_MONSTER_ID) as number),
+        name: s.params.get(PARAM_ID.CA_NAME) as string,
+        description: s.params.get(PARAM_ID.CA_DESCRIPTION) as string,
+        category: categoryEnum.map.get(categoryValue) || 'Unknown',
+        tier: this.getLegacyTier(s.params.get(PARAM_ID.CA_TIER_ID) as number),
+        clientSortId: '' + i,
+      };
+    });
+  
     if (options.json) {
       this.writeToFile(allTasksFormatted, 'combat.json');
     } else {
       console.log(JSON.stringify(allTasksFormatted, replacer));
     }
-    return allTasksFormatted;
+  
+    return allTasksFormatted; 
   }
+  
+  
 
   public async handleLeagues4(options: any): Promise<ITask[]> {
-    const structId: ParamID = 873 as ParamID;
+    // console.log('handleLeagues4 invoked with options:', options);
+  
+    const structId: ParamID = PARAM_ID.LEAGUE_VARBIT_INDEX;
+    const categoryParamId: ParamID = PARAM_ID.LEAGUE_CATEGORY_ID;
+    const tierParamId: ParamID = PARAM_ID.LEAGUE_TIER_ID;
+  
+    // Define tierMap (could probably move this not here)
+    const tierMap = {
```

Comment:
There's actually a cache enum for this if you're feeling intrepid. I think I reference it in task-json-store?

Reply:
Directly lifted the actual values from the task-json-store, but standardised it within param-ids.ts